esp-hal/src/dma/mod.rs

```diff
@@ -366,6 +366,26 @@ impl DmaDescriptor {
             true => Owner::Dma,
         }
     }
+
+    fn iter(&self) -> impl Iterator<Item = &DmaDescriptor> {
+        core::iter::successors(Some(self), |d| {
+            if d.next.is_null() {
+                None
+            } else {
+                Some(unsafe { &*d.next })
+            }
+        })
+    }
+
+    fn iter_mut(&mut self) -> impl Iterator<Item = &mut DmaDescriptor> {
+        core::iter::successors(Some(self), |d| {
+            if d.next.is_null() {
+                None
+            } else {
+                Some(unsafe { &mut *d.next })
+            }
+        })
+    }
```

Comment:
Are these still used? They should be marked `unsafe`.

I want to also note that the last time I used this in my project, it was noticeably slower (less FPS) that iterating through the slice.